File: src/engine_config.cpp
#include "engine_config.hpp"

struct engine_config_s get_engine_config(lua_State *L, sf::String filename){
    debug("Fetching engine config from %s", filename.toAnsiString().c_str());
    scripting_dofile(L, filename);
    
    struct engine_config_s engine_config;
    
    engine_config.screen_width = scripting_get_global_int(L, "SCREEN_WIDTH");
    engine_config.screen_height = scripting_get_global_int(L, "SCREEN_HEIGHT");
    engine_config.maximum_framerate = scripting_get_global_int(L, "MAXIMUM_FRAMERATE");
    
    return engine_config;
}

File: deucalion-rs/src/engine_config.rs
//! Functions for managing and acquiring engine configuration

// TODO: Make this work with the Lua subsystem.

/// A datastructure containing configuration details for the engine
#[derive(PartialEq, Eq, Debug)]
pub struct EngineConfig {
    /// The width of the window in which the engine will run.
    pub screen_width: u32,
    /// The height of the window in which the engine will run.
    pub screen_height: u32,
    /// The maximum framerate at which the engine should attempt to run.
    pub maximum_framerate: u32,
}

/// Acquire the engine's configuration. Currently this is arbitrary defaults; in
/// the future, it will be from a Lua script.
pub fn get_engine_config() -> Result<EngineConfig, String> {
    Ok(EngineConfig {
        screen_width: 640,
        screen_height: 480,
        maximum_framerate: 60,
    })
}
